heltour/tournament/models.py

```diff
@@ -1485,6 +1501,28 @@ def other_seasons(self):
     def player(self):
         return Player.objects.filter(lichess_username__iexact=self.lichess_username).first()
 
+    @classmethod
+    def can_register(cls, user, season):
+        if not season.registration_open:
+            return False
+        try:
+            return cls.objects.get(lichess_username=user.username, season=season).status != 'rejected'
+        except cls.DoesNotExist:
+            return True
+
+
+    @classmethod
+    def get_registration(cls, user, season):
+        try:
+            r = cls.objects.get(lichess_username=user.username, season=season)
```

Comment:
We should probably deal with multipleobjects as well in this case. For old seasons there will be be duplicate registrations, and if this gets called on them at any point, then we'll need to deal with it.  

I'm not sure what we should do in that case. An alternative approach is to make a migration that removes old duplicate registrations. But I'm not wild about that idea.

Reply:
I can catch it.  I tend to dislike catching things that are truly errors though.  I'm thinking thats probably not best practice in a web server though.  But we would need to report that exception in some way that blaring alarms go off for us since this should never happen.

I feel like multiple registrations really should be a gross violation and be purged/prevented.